app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt

```diff
@@ -242,11 +245,44 @@ class StoryFragmentTest {
     assertThat(storyToolbarTitle.textAlignment).isEqualTo(TEXT_ALIGNMENT_VIEW_START)
   }
 
+  // its view
+  @Test // TODO(#4212): Error -> Only the original thread that created a view hierarchy can touch
+  fun testStoryFragment_toolbarTitle_readerOn_marqueeInRtl_isDisplayedCorrectly() {
+    activityTestRule.launchActivity(createFractionsStoryActivityIntent())
+    testCoroutineDispatchers.runCurrent()
+    accessibilityService.setScreenReaderEnabled(true)
+
+    val storyToolbarTitle: TextView =
+      activityTestRule.activity.findViewById(R.id.story_toolbar_title)
+    ViewCompat.setLayoutDirection(storyToolbarTitle, ViewCompat.LAYOUT_DIRECTION_RTL)
+
+    onView(withId(R.id.story_toolbar_title)).perform(click())
+    assertThat(storyToolbarTitle.isFocused).isEqualTo(false)
+    assertThat(storyToolbarTitle.textAlignment).isEqualTo(TEXT_ALIGNMENT_VIEW_START)
+  }
+
+  // its view
   @Test // TODO(#4212): Error -> Only the original thread that created a view hierarchy can touch
+  fun testStoryFragment_toolbarTitle_readerOff_marqueeInLtr_isDisplayedCorrectly() {
+    activityTestRule.launchActivity(createFractionsStoryActivityIntent())
+    testCoroutineDispatchers.runCurrent()
+    accessibilityService.setScreenReaderEnabled(false)
+
+    val storyToolbarTitle: TextView =
+      activityTestRule.activity.findViewById(R.id.story_toolbar_title)
+    ViewCompat.setLayoutDirection(storyToolbarTitle, ViewCompat.LAYOUT_DIRECTION_LTR)
+
+    onView(withId(R.id.story_toolbar_title)).perform(click())
+    assertThat(storyToolbarTitle.isFocused).isEqualTo(false)
+    assertThat(storyToolbarTitle.textAlignment).isEqualTo(TEXT_ALIGNMENT_VIEW_START)
+  }
+
   // its view
-  fun testStoryFragment_toolbarTitle_marqueeInLtr_isDisplayedCorrectly() {
+  @Test // TODO(#4212): Error -> Only the original thread that created a view hierarchy can touch
+  fun testStoryFragment_toolbarTitle_readerOn_marqueeInLtr_isDisplayedCorrectly() {
     activityTestRule.launchActivity(createFractionsStoryActivityIntent())
     testCoroutineDispatchers.runCurrent()
+    accessibilityService.setScreenReaderEnabled(false)
```

Comment:
The test name implies that the reader should be on, not off.

Reply:
It looks like you may have mistakenly switched this test and the one above, because they are both asserting behaviour opposite of their test names.